### PR TITLE
[bitnami/kube-prometheus]: add prometheus config-reloader servicemonitor

### PR DESCRIPTION
**Pull request by jaygridley**
### Description of the change

- add Service and ServiceMonitor for Prometheus config-reloader sidecar container

### Applicable issues

- fixes https://github.com/bitnami/charts/issues/15765

### Additional information

Need to be tested

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kube-prometheus/Chart.yaml [hunk 1]
```diff
@@ -35,4 +35,4 @@ sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/prometheus
   - https://github.com/bitnami/containers/tree/main/bitnami/alertmanager
   - https://github.com/prometheus-operator/kube-prometheus
-version: 8.4.1
+version: 8.6.0
```
Comment (aoterolorenzo) [inline bitnami/kube-prometheus/Chart.yaml:38]: Bumped 2 minors here

```suggestion
version: 8.5.0
```

File: bitnami/kube-prometheus/templates/prometheus/config-reloader-service.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{{- if and .Values.prometheus.enabled .Values.prometheus.configReloader.serviceMonitor.enabled }}
```
Comment (aoterolorenzo) [inline bitnami/kube-prometheus/templates/prometheus/config-reloader-service.yaml:1]: I think the service shouldn't depend on the serviceMonitor.enabled value, but in an apart service.enabled value:

```
prometheus:
    configReloader:
        service:
            enabled: true
            serviceMonitor:
                enabled: true
```

```suggestion
{{- if and .Values.prometheus.enabled .Values.prometheus.configReloader.service.enabled }}
```

**Review by aoterolorenzo — CHANGES_REQUESTED** [review]
Hi @jaygridley 

I left some comments above.

On the other hand, all seems to be working properly

```
lorenzoa@lorenzoaM79V2 kube-prometheus % sudo kubectl port-forward --namespace default svc/kube-prometheus-prometheus-config-reloader 8080
lorenzoa@lorenzoaM79V2 kube-prometheus % telnet 127.0.0.1 8080
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
```

File: bitnami/kube-prometheus/templates/prometheus/config-reloader-service.yaml (hunk 2 above)
Comment (jaygridley) [reply bitnami/kube-prometheus/templates/prometheus/config-reloader-service.yaml:1]: Do we want to have both Service and ServiceMonitor to be enabled by default?

File: bitnami/kube-prometheus/values.yaml [hunk 3]
```diff
@@ -1404,6 +1404,96 @@ prometheus:
...
+      ## If "ClientIP", consecutive client requests will be directed to the same Pod
+      ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+      ##
+      sessionAffinity: None
+      ## @param prometheus.configReloader.service.sessionAffinityConfig Additional settings for the sessionAffinity
+      ## sessionAffinityConfig:
+      ##   clientIP:
+      ##     timeoutSeconds: 300
+      ##
+      sessionAffinityConfig: {}
```
Comment (CeliaGMqrz) [inline bitnami/kube-prometheus/values.yaml:1474]: Please, follow the same structure as the rest of the parameters with "##"

Comment (jaygridley) [reply bitnami/kube-prometheus/values.yaml:1474]: @CeliaGMqrz I am kind of struggling to understand what is needed here. IMHO the structure is already the same as for other `sessionAffinityConfig: {}` values within the file. Could you please add more details?

File: bitnami/kube-prometheus/values.yaml [hunk 4]
```diff
@@ -1404,6 +1404,96 @@ prometheus:
...
+    serviceMonitor:
```
Comment (CeliaGMqrz) [inline bitnami/kube-prometheus/values.yaml:1475]: Sorry, I wanted to say something like this, to keep the same structure for comments:
```suggestion
      ##
      sessionAffinityConfig: {}
    ## Create a servicemonitor for the configReloader
    ##
    serviceMonitor:
```

File: bitnami/kube-prometheus/templates/prometheus/config-reloader-service.yaml (hunk 2 above)
Comment (CeliaGMqrz) [reply bitnami/kube-prometheus/templates/prometheus/config-reloader-service.yaml]: I think we should not leave both by default.

**Review by CeliaGMqrz — COMMENTED** [review]
Hi, sorry for the delay. I have left some comments.

File: bitnami/kube-prometheus/Chart.yaml (hunk 1 above)
Comment (CeliaGMqrz) [inline bitnami/kube-prometheus/Chart.yaml:38]: Sorry, we have new changes, can you please update the chart version again?
```suggestion
version: 8.7.0
```

File: bitnami/kube-prometheus/Chart.yaml [hunk 5]
```diff
@@ -35,4 +35,4 @@ sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/prometheus
   - https://github.com/bitnami/containers/tree/main/bitnami/alertmanager
   - https://github.com/prometheus-operator/kube-prometheus
-version: 8.6.0
+version: 8.7.0
```
Comment (CeliaGMqrz) [inline bitnami/kube-prometheus/Chart.yaml:38]:
```suggestion
version: 8.8.0
```

**Review by CeliaGMqrz — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**